myosuite/envs/myo/myochallenge/run_track_v0.py

```diff
@@ -0,0 +1,962 @@
+                                'l_foot_relative_X', 'l_foot_relative_Y', 'l_foot_relative_Z', 
+                                'r_foot_relative_X', 'r_foot_relative_Y', 'r_foot_relative_Z',
+                                'pelvis_vel_X', 'pelvis_vel_Y', 'pelvis_vel_Z']:
+                self.init_qpos[self.sim.model.joint(joint).qposadr[0]] = self.INIT_DATA[start_idx,self.imitation_lookup[joint]]
+
+        # Get the Yaw from the init pose
+        default_quat = self.init_qpos[3:7].copy() # Get the default facing direction first
+
+        init_quat = self.intrinsic_EulerXYZ_toQuat(self.INIT_DATA[start_idx, self.imitation_lookup['pelvis_euler_roll']], 
+                                                   self.INIT_DATA[start_idx, self.imitation_lookup['pelvis_euler_pitch']], 
+                                                   self.INIT_DATA[start_idx, self.imitation_lookup['pelvis_euler_yaw']])
+        self.init_qpos[3:7] = init_quat
+        
+        # Use the default facing direction to set the world frame velocity
+        temp_euler = self.get_intrinsic_EulerXYZ(default_quat)
+        world_vel_X, world_vel_Y = self.rotate_frame(self.INIT_DATA[start_idx, self.imitation_lookup['pelvis_vel_X']], 
+                                                     self.INIT_DATA[start_idx, self.imitation_lookup['pelvis_vel_Y']],
+                                                     temp_euler[2])
+        self.init_qvel[:] = 0
+        self.init_qvel[0] = world_vel_X
+        self.init_qvel[1] = world_vel_Y
+        self.init_qvel[2] = self.INIT_DATA[start_idx, self.imitation_lookup['pelvis_vel_Z']]
+
+        state = self.init_lookup[start_idx]
+        # Override previous OSL controller
+        self.OSL_FSM = self.build_4_state_FSM(self.OSL, state)
+
+    def adjust_model_height(self):
+
+        curr_qpos = self.sim.data.qpos.copy()
+        curr_qvel = self.sim.data.qvel.copy()
+
+        temp_sens_height = 100
+        for sens_site in ['r_heel_btm', 'r_toe_btm', 'l_heel_btm', 'l_toe_btm']:
+            if temp_sens_height > self.sim.data.site(sens_site).xpos[2]:
+                temp_sens_height = self.sim.data.site(sens_site).xpos[2].copy()
+
+        diff_height = 0.0 - temp_sens_height
+        curr_qpos[2] = curr_qpos[2] + diff_height
+
+        return curr_qpos, curr_qvel
+
+    def generate_init_lookup(self, keys, value, existing_dict=None):
+        # Initialize an empty dictionary if existing_dict is None
+        if existing_dict is None:
+            result_dict = {}
+        else:
+            # Use the existing dictionary if provided
+            result_dict = copy.deepcopy(existing_dict)
+
+        # Iterate through the list of integers (keys)
+        for key in keys:
+            # Assign each key in the dictionary with the value "early_stance"
+            result_dict[key] = value
+
+        return result_dict
+
+    """
+    OSL related functions
```

Comment:
Break it into its own class. This should be extendible. Others should be able to leverage it and build forward.

Reply:
Separated in its own class